_posts/2021-12-16-new-security-features-wildfly.adoc

```diff
@@ -0,0 +1,59 @@
+---
+layout: post
+title: 'New Security Features in WildFly'
+date: 2021-12-16
+tags: release jaas security-realm oidc filesystem-realm
+synopsis: An overview of some new security features in WildFly.
+author: fjuma
+---
+
+Since WildFly 26 was just released, we wanted to highlight a new security feature included
+in this release and a couple other security features as well.
+
+== Support for JAAS Security Realms in WildFly 26
+
+WildFly 26 now adds support for JAAS security realms to the Elytron subsystem. This means that
+it's possible to use custom LoginModules for authentication and authorization. More information,
+including an example on how to quickly get started with this new feature, can be found in this
+https://wildfly-security.github.io/wildfly-elytron/blog/jaas-realm/[blog post].
+
+== Support for OpenID Connect
+
+WildFly 25 added the ability to secure applications deployed to WildFly with OpenID Connect,
+without needing to use the Keycloak client adapter. For more details about this recent
+feature and an example that shows how to secure an application deployed to WildFly with
+the Keycloak OpenID provider, take a look at this
+https://wildfly-security.github.io/wildfly-elytron/blog/securing-wildfly-apps-openid-connect/[blog post].
+
+== Upcoming Filesystem Realm Enhancements
+
+We will be adding the ability to encrypt the identities that are stored in Elytron's filesystem-based
+security realm. In addition to this, we will be adding the ability to verify the integrity of these
+identities as well. For all the details on this upcoming feature, check out this https://wildfly-security.github.io/wildfly-elytron/blog/upcoming-filesystem-encryption-integrity[blog post].
+
```

Comment:
@fjuma 
== Upcoming client side default SSL context provider

We will be adding a Java security provider that loads Elytron client configuration and registers default SSL context for the JVM. When this provider is registered with high enough priority, client libraries that use `SSLContext.getDefault()` will obtain SSL context instantiated and configured by Elytron client. For all the details on this upcoming feature, check out this 
https://wildfly-security.github.io/wildfly-elytron/blog/client-side-jvm-wide-default-sslcontext/[blog post].